test/factory.js

```diff
@@ -59,10 +59,48 @@ YUI(GlobalConfig).add('juju-tests-factory', function(Y) {
       return fakeStore;
     },
 
+    _fetchCharmData: function() {
```

Comment:
this says 'fetchCharmData' which is singular but it's getting a list of results back. I'm confused as to the purpose of this factory method. Maybe some docs on it and naming updates might help make it more clear?